Add benchmarks for scaled_mm and scaled_grouped_mm

jainapurva opened the pull request:
### Summary
Add operator benchmarks for scaled_mm and scaled_grouped_mm operations
This PR introduces comprehensive microbenchmarks for FP8 scaled matrix multiplication operations, targeting large language model inference and training workloads.

### What's Added
**New file:** benchmarks/operator_benchmark/pt/scaled_mm_test.py; benchmarks/operator_benchmark/pt/scaled_grouped_mm_test.py
**Two benchmark classes:**
1. ScaledMMBenchmark - Tests torch.nn.functional.scaled_mm for standard FP8 matrix multiplication
2. ScaledGroupedMMBenchmark - Tests torch.nn.functional.scaled_grouped_mm for MoE (Mixture of Experts) grouped matmul operations

### Model Shapes
Benchmarks cover realistic production shapes from two major model architectures:
- **DeepSeek-V3 671B:** Matrix dimensions (M, N=2048, K=7168)
- **Llama4 16e:** Matrix dimensions (M, N=8192, K=5120)

With varying batch sizes (M=16384, 128000) and expert counts (G=1,2,4,8) for grouped operations.

jainapurva committed:
Add benchmarks for scaled_mm and scaled_grouped_mm

jainapurva committed:
benchmarks: update scaled_mm(+grouped) long configs

jainapurva committed:
updates

jainapurva committed:
Merge remote-tracking branch 'origin/main' into scaled_mm1

jainapurva committed:
updates

jainapurva committed:
Merge remote-tracking branch 'origin/main' into scaled_mm1

jainapurva committed:
updates

slayton58 reviewed (approved):
LGTM - few nits, nothing major

jainapurva committed:
Refactor scaled_mm benchmarks to reduce code duplication

Created scaled_mm_common.py to consolidate shared utilities between
scaled_mm_test.py and scaled_grouped_mm_test.py benchmarks:

- Moved get_test_scaled_matmul_cuda() helper to common file
- Moved get_float8_dtype() helper to common file
- Moved supports_fp8_deepseek_blockwise_scaling() helper to common file
- Extracted SCALED_MM_BASE_SHAPES for shared benchmark configurations
- Refactored duplicate FP8 blockwise initialization methods

This reduces code duplication by 121 lines (36% reduction) and ensures
both benchmark files use consistent configurations and utilities.

jainapurva committed:
Apply formatting and linting fixes to scaled_mm benchmarks

- Reorder and alphabetize imports per Python style guidelines
- Fix import grouping (stdlib, third-party, local)
- Remove extra blank lines
- Fix multi-line function argument formatting
- Convert tuple comprehension to list comprehension for MNKG_list

jainapurva committed:
Add smaller benchmark shapes for faster testing

Added 3 smaller matrix shapes to SCALED_MM_BASE_SHAPES:
- 1024 × 1024 × 1024 (classic small square)
- 2048 × 4096 × 2048 (medium rectangular)
- 4096 × 2048 × 4096 (medium with different aspect ratio)

These smaller shapes enable faster benchmarking and are automatically
picked up by both scaled_mm_test.py and scaled_grouped_mm_test.py.

For scaled_grouped_mm, the first 2 small shapes will be tested with
all group counts [1, 2, 4, 8], while larger shapes use [1, 8] only.

jainapurva committed:
Enable MX blockwise scaling support for HIP/ROCm in benchmarks

Removes the CUDA-only restriction for MX (mxfp8/mxfp4) benchmarks and
enables them to run on HIP/ROCm platforms by using SwizzleType.NO_SWIZZLE
instead of SWIZZLE_32_4_4.

Changes:
- scaled_mm_test.py: Add conditional swizzle type selection (NO_SWIZZLE
  for HIP, SWIZZLE_32_4_4 for CUDA) in _init_mx_blockwise()
- scaled_grouped_mm_test.py: Delegate swizzle selection to
  _build_scaled_grouped_mm_kwargs() helper which already handles
  HIP vs CUDA correctly
- Update config generation to enable MX benchmarks on both CUDA and HIP
- Keep NVFP4 as CUDA-only (not supported on HIP)

This matches the pattern used in test/test_scaled_matmul_cuda.py where
the _build_scaled_grouped_mm_kwargs() helper automatically selects the
correct swizzle type based on platform.

jainapurva committed:
ci runner

jainapurva committed:
test push

jainapurva committed:
<Replace this line with a title. Use 1 line only, 67 chars or less>

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

jainapurva committed:
updates

jainapurva committed:
Expand benchmark tests to include additional operations

jainapurva commented:
@pytorchmergebot merge -i

danielvegamyhre commented on benchmarks/operator_benchmark/pt/scaled_grouped_mm_test.py:
```diff
@@ -0,0 +1,334 @@
+    (16384, 2048, 7168, 1),
+    (16384, 2048, 7168, 8),
+    (128000, 2048, 7168, 1),
+    (128000, 2048, 7168, 8),
```
sorry for the insanely late reply, not sure how i missed this, will check my notifications for pytorch/pytorch - but yes, these shapes are fine.

looks like dsv3 671b and llama4 shapes for routed expert dim/hidden_dim, and for the leading/batch dim, 32k-128k or so is what i generally test. 

for MoE training people often use dp2ep parallelism and try to have a large "total M" dim to improve GEMM efficiency with these small/granular experts used by the dsv3/qwen/kimi/etc style models. so i focus on that.
